controllers/postgresql_controller.go

```diff
@@ -94,3 +100,59 @@ func (a *postgreSQLAdapter) getServiceType() string {
 func (a *postgreSQLAdapter) getDiskSpace() string {
 	return a.Spec.DiskSpace
 }
+
+func (a *postgreSQLAdapter) performUpgradeTaskIfNeeded(ctx context.Context, avn *aiven.Client, avnGen avngen.Client, old *aiven.Service) error {
+	var currentVersion string = old.UserConfig["pg_version"].(string)
+	targetUserConfig := a.getUserConfig().(*pguserconfig.PgUserConfig)
+	if targetUserConfig == nil {
+		return nil
+	}
+	var targetVersion *string = targetUserConfig.PgVersion
+
+	// No need to upgrade if pg_version hasn't changed
+	if targetVersion == nil || *targetVersion == currentVersion {
+		return nil
+	}
```

Comment:
I think it better to not use a pointer string here.

```suggestion
	if targetUserConfig == nil || targetUserConfig.PgVersion == nil{
		return nil
	}

	// No need to upgrade if pg_version hasn't changed
	targetVersion := *targetUserConfig.PgVersion
	if targetVersion == currentVersion {
		return nil
	}
```